api/discord/bot.php

```diff
@@ -63,12 +63,15 @@ public static function send_message($message, $channel_id, $attachments = null)
 
 		// Add files to array
 		$files = array();
+        /** @noinspection PhpArrayUsedOnlyForWriteInspection */
+        $tmp_files = [];    // needed if more than 1 attachment to prevent gc for deleting the temp files

```

Comment:
This is slightly dangerous because a _good_ rubbish (garbage) collector should clean up variables that are unused, and I wouldn't be surprised if in the future the gc was updated to do exactly that.

Instead, we should just initialise a `$tmp_files` array by creating `len($attachments)` temp files before we enter into the attachments foreach. Then, inside of the foreach, we can do `$file = $tmp_files[$k]`

Reply:
Removed the $files var. I didn't make `len($attachments)` temp files since we don't need a temp file for [files already on the server](https://github.com/untrobotics/website/pull/166/files#diff-66182fd563126a5c33c16c088823766d47ad57fff98a6a755b260e23e1e56d60R78-R83).

The temp files are inserted into the array at `$k`.